specification/quota/resource-manager/Microsoft.Quota/preview/2021-03-15-preview/examples/putComputeOneSkuQuotaRequest.json

```diff
@@ -15,29 +18,37 @@
   "responses": {
     "202": {
       "headers": {
-        "location": "https://management.azure.com/subscriptions/D7EC67B3-7657-4966-BFFC-41EFD36BAAB3/providers/Microsoft.Compute/locations/eastus/providers/Microsoft.Quota/quotaLimitsRequests/2B5C8515-37D8-4B6A-879B-CD641A2CF605?api-version=2021-03-15-preview",
         "Retry-After": "30",
         "Azure-AsyncOperation": "https://management.azure.com/subscriptions/D7EC67B3-7657-4966-BFFC-41EFD36BAAB3/providers/Microsoft.Compute/locations/eastus/providers/Microsoft.Quota/operationsStatus/2B5C8515-37D8-4B6A-879B-CD641A2CF605?api-version=2021-03-15-preview"
       },
       "body": {
-        "id": "/subscriptions/D7EC67B3-7657-4966-BFFC-41EFD36BAAB3/providers/Microsoft.Compute/locations/eastus/providers/Microsoft.Quota/quotaLimits/standardFSv2Family",
-        "type": "Microsoft.Quota/quotaLimits",
+        "id": "/subscriptions/D7EC67B3-7657-4966-BFFC-41EFD36BAAB3/providers/Microsoft.Compute/locations/eastus/providers/Microsoft.Quota/quotas/standardFSv2Family",
+        "type": "Microsoft.Quota/quotas",
         "name": "standardFSv2Family",
         "properties": {
-          "limit": 200,
+          "limit": {
+            "limitObjectType": "LimitValue",
+            "value": 10
+          },
           "name": {
             "value": "standardFSv2Family"
           }
         }
       }
     },
     "200": {
+      "headers": {
+        "location": "https://management.azure.com/subscriptions/D7EC67B3-7657-4966-BFFC-41EFD36BAAB3/providers/Microsoft.Compute/locations/eastus/providers/Microsoft.Quota/quotaRequests/2B5C8515-37D8-4B6A-879B-CD641A2CF605?api-version=2021-03-15-preview"
```

Comment:
I believe I commented on this in a previous PR but a "location" header should keep returning 202 until the async operation is complete. If you point at quotaRequests they will get a 200 back with a quotaRequests resource immediately and think the quota operation is complete.

Reply:
I will put the operations url for the location header, which will return 202, until the operation is completed.

Reply:
yes, from an async perspective that is fine.  I think you still have a discoverability problem here.  How does a user discover the quotaRequests resource related to the PUT they did on a quotas resource?  There should be some tie between the two.  Maybe as simple as referencing the "quotas" resource ID that a quotaRequests is related to in the quotaRequests response body (and allowing filtering on that property in the collection GET)

Reply:
The way it's implemented -
1. User calls the Put API, return is 202 with the Azure-AsyncOperation" : operations url.
Operations API processing:
1. If request is in progress, it returns "Azure-AsyncOperation" with operations api.
2.  User or SDK keeps on checking the operationsStatus API, based on retry-after period. 
3. When the provisioning state is terminal state - failed or succeeded. In the response the "location" header is set to the quotaLimitRequests and  "Azure-AsyncOperation" header is not set.
4. The SDK and ARM template calls the url in location for the quotaLimitRequests and displays the result based on provisioning State. 

I don't see a case, where the PUT or patch api is going to return 200 for long running operation.

Reply:
At the end of an async operation the "location" header must point to the result of the request.  Showing a different resource when the user did a PUT on a quotas resource violates that (see # 5 here:  https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#202-accepted-and-location-headers).  202 is fine for PATCH but 200/201 + provisioningState + azure-asyncoperation is the preferred model for async PUTs (https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#creatingupdating-using-put).  

You CAN put the resource ID of the quotaRequests resource in the "properties" of the azure-asyncoperation operationStatuses resource but I don't know how discoverable that is.  You could try it and see what that experience is like (https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#azure-asyncoperation-resource-format)

Reply:
The operations status is following the guidelines at - https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#azure-asyncoperation-resource-format.

"id": "/subscriptions/id/locations/westus/operationsStatuses/sameguid - is the requestId. Which returns the-
"status" : "RP defined values | Succeeded | Failed | Canceled",
Request details  in properties: {what quota was requested}
In case of error, error details in -
error" : {
    /\* RP must return the *code* and *messages* fields. Please use the schema for the "ErrorResponse" Type from the Common Types definition in the Azure Rest API Specifications [repository](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/common-types/resource-management/v2/types.json) \*/
    "code": "BadArgument",
    "message": "The provided database &#39;foo&#39; has an invalid username."
  }

The ARM template and Azure SDK works fine, so looks like we are following the right guidelines for operationsStatus.
Example from ARM template deployment:
VERBOSE: 10:57:23 AM - Resource Microsoft.Quota/quotaLimits 'PublicIPAddresses' provisioning status is running
And finally it shows Failed (with error details) or Succeeded.

Does it look fine?

Reply:
You are combining a few async patterns into one but not following the requirements for any explicitly.  If you are returning a 202 you MUST return a location header (https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#creating-or-updating-resources-asynchronously).  You can ALSO return an azure-asyncoperation header if you wish but that is secondary.  ARM's async operation tracking code is more lenient than the spec but the spec must be followed so users that are tracking it using their own code that is explicitly following the spec don't fail

Reply:
I have updated the 202 response and headers based on - https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#creating-or-updating-resources-asynchronously.

Only location header is returned, which points to operationsStatus for 202 and quotaRequests for 200.
When the operation reaches the terminal state, then the location header will point to quotaRequests URL, which will have all the error details or success message.